src/System.Text.RegularExpressions/tests/GroupCollectionReadOnlyDictionaryTests.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public static class GroupCollectionReadOnlyDictionaryTests
+    {
+        [Fact]
+        public static void IReadOnlyDictionary_TryGetValueSuccess()
+        {
+            Regex regex = new Regex(@"(?<A1>a*)(?<A2>b*)(?<A3>c*)");
+            Match match = regex.Match("aaabbccccccccccaaaabc");
+
+            var groups = match.Groups;
+
+            if (groups.TryGetValue("A1", out Group value))
+            {
+                Assert.Equal("aaa", value?.Value);
+            }
+            else
+            {
+                Assert.True(false, "Value should exist");
```

Comment:
Does xUnit not have an [Assert.Fail](https://msdn.microsoft.com/en-us/library/microsoft.visualstudio.testtools.unittesting.assert.fail.aspx) method?

Reply:
I couldn't find it. See also https://stackoverflow.com/questions/14631923/xunit-net-cannot-find-assert-fail-and-assert-pass-or-equivalent

Reply:
I couldn't find it. See also https://stackoverflow.com/questions/14631923/xunit-net-cannot-find-assert-fail-and-assert-pass-or-equivalent

Reply:
Ah, interesting.  @ViktorHofer do you know if there's a reason why it's not there?  Should I open an issue as a feature request for xUnit?

Reply:
xunit has rejected this specific api proposal couple of times. We settled with just doing it this way...

Reply:
Feel free to post on xunit repo though

Reply:
Why write it with this if/else and Assert.True(false, ...) rather than just doing:
```C#
Assert.True(groups.TryGetValue("A1", out Group value));
Assert.Equal("aaa", value.Value);
```
?